openingd/openingd.c

```diff
@@ -463,6 +469,12 @@ static bool setup_channel_funder(struct state *state)
 	 * until we know their funding_pubkey) */
 	temporary_channel_id(&state->channel_id);
 
+#if DEVELOPER
+	/* --dev-force-tmp-channel-id specified */
+	if (dev_force_tmp_channel_id)
+		memcpy(&state->channel_id, dev_force_tmp_channel_id,
+		       sizeof(*dev_force_tmp_channel_id));
```

Comment:
`state->channel_id = *dev_force_tmp_channel_id;`

People often forget that struct assignment works (whereas comparison doesn't) in C.

Reply:
oh. good to know.